docs/time/performance.inc.rst

```diff
@@ -10,3 +10,76 @@
 ..
 .. Here we provide some tips and tricks for how to optimize performance of code
 .. using `astropy.time`.
+
+Light Travel Times
+------------------
+
+Listed below are two approaches to calculating light travel times for tens of
+thousands of sources in a degree patch of the sky. The first approach calculates
+the travel times without iteration while the second calculates using iteration.
+The second approach using iteration should be avoided as it is extraordinarily slow.
+
+.. note::
+   The wall times and cpu times were calculated on google co-laboratory.
+
+   >>> import numpy as np
+   >>> import astropy.coordinates as coord
+   >>> import astropy.units as u
+   >>> from astropy.time import Time
+
+   >>> ra = np.random.normal(0.0, 1.0, 10000)
+   >>> dec = np.random.normal(0.0, 1.0, 10000)
+
+   >>> coos = coord.SkyCoord(ra, dec, unit=u.deg)
+   >>> observatory = coord.EarthLocation.from_geocentric(5327448.9957829, -1718665.73869569, 3051566.90295403, unit='m')
+   >>> ltts = [Time.light_travel_time(self=Time.now(), skycoord=coos, location=observatory) for coo in coos]
+
+
+If you have an internet connection, **coord.EarthLocation.of_site('lapalma')** can be used.
+However, if that does not work you, or you don't have an internet connection, use
+**coord.EarthLocation.from_geocentric(5327448.9957829, -1718665.73869569, 3051566.90295403, unit='m')**
+
+Approach Not Using Iteration
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+This approach takes advantage of the vectorized operations in numpy.
+
+   >>> ltt = Time.light_travel_time(self=Time.now(), skycoord=coos, location=observatory)
+
+   **CPU times:** user 31.4 ms, sys: 1 ms, total: 32.4 ms
+
+   **Wall time:** 31.8 ms
+
+Approach Using Iteration
+^^^^^^^^^^^^^^^^^^^^^^^^
+Iterating over a SkyCoord object is not fast, due to the how a SkyCoord is constructed. Therefore
+the user time is significantly longer.
+
+   >>> ltts = [Time.light_travel_time(self=Time.now(),skycoord=coos, location=observatory) for coo in coos]
```

Comment:
```suggestion
   >>> ltts = [Time.light_travel_time(self=Time.now(), skycoord=coo, location=observatory) for coo in coos]
```
It looks like the code is mistakenly converting all of the coordinates with each iteration.  The time measurements should be redone.

Reply:
It would probably make sense to avoid doing ``Time.now()`` for each item?

Reply:
^ right, you'd probably want to calculate all the times in a single batch and access each relevant one with each of the iterations. That redundant time calculation is probably what's slowing this down massively.